week-1-ts-ecosystem/w01e03-object-oriented-structure/src/modules/cart/cart.module.ts

```diff
@@ -0,0 +1,64 @@
+import { Product, ProductType } from "../product/product.module";
+
+type currency = "PLN" | "USD" | "EUR";
+export class ShoppingCart {
+  #products: Product[] = [];
+  #name: string;
+  #currency: currency;
+  #total: number = 0;
+  #basketType: ProductType;
+  constructor(name: string, currency: currency, basketType: ProductType) {
+    this.#name = name;
+    this.#currency = currency;
+    this.#basketType = basketType;
+  }
+  addProduct(product: Product) {
+    if (product.type !== this.#basketType) {
+      console.warn("Product type and basket type are not matching");
+    } else {
+      this.#products.push(product);
+      this.countTotal();
+    }
+  }
+  removeProduct(productToRemove: Product) {
+    const filteredProducts = this.#products.filter(
+      (product) => product.id !== productToRemove.id
+    );
+    this.#products = filteredProducts;
+    this.countTotal();
+  }
+
+  countTotal() {
+    this.#total = this.#products.reduce(
```

Comment:
Zobacz... żeby wiedzieć jaki jest `#total` muszę za każdym razem wiedzieć że trzeba odpalić `countTotal()` a co jak zapomnę tego dopisać w kodzie przy operacjach `add` czy `remove`

A co jakby `#total` było wyliczane - za każdym razem jak o to zapytam?

Poczytaj proszę o `getterach` - to Cię powinno naprowadzić.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get